pyomo/repn/plugins/nl_writer.py

```diff
@@ -1790,21 +1796,27 @@ def _linear_presolve(self, comp_by_linear_var, lcon_by_linear_nnz, var_bounds):
                     b,
                 )
                 # Tighten variable bounds
-                x_lb, x_ub = var_bounds[x]
                 lb, ub = var_bounds[_id]
-                if lb is not None:
-                    lb = (lb - b) / a
-                if ub is not None:
-                    ub = (ub - b) / a
-                if a < 0:
-                    lb, ub = ub, lb
-                if x_lb is None or (lb is not None and lb > x_lb):
-                    x_lb = lb
-                if x_ub is None or (ub is not None and ub < x_ub):
-                    x_ub = ub
-                var_bounds[x] = x_lb, x_ub
-                if x_lb == x_ub and x_lb is not None:
-                    fixed_vars.append(x)
+                if a == 0:
+                    if (lb is not None and lb > b) or (ub is not None and ub < b):
+                        raise InfeasibleConstraintException(
+                            f'Model is infeasible: {lb} <= {var_map[_id]} <= {ub} and {var_map[_id]} == {a}*{var_map[x]} + {b} cannot both be satisfied'
+                        )
```

Comment:
If the model is feasible, shouldn't this end up fixing `var_map[_id]`?  Maybe the logic for trapping a == 0 should be moved earlier (to new line 1784?)?

Reply:
Ahh. I see. I missed that logic. Good catch. I'll update.